allPass.ts

```diff
@@ -0,0 +1,25 @@
+import curryN from "./utils/curry_n.ts"
+import { Curry1, Func } from "./utils/types.ts"
+import { max } from './max.ts'
+import { pluck } from './pluck.ts'
+import { reduce } from './reduce.ts'
+
+
+function _allPass(preds: any) {
+  return curryN(reduce(max, 0, pluck('length', preds)), function(this: any) {
```

Comment:
can't we have this function more readable, like get the length of function first, the declare the function and then curry it and return

Reply:
Yes, but since length is being used only once, declaring separately will just increase the number of lines.

Reply:
yeh, but that increases the readability. We have our code more than 8000 lines, which is enough for a big project. So, an extra variable is not the pain. Developing time is more precious than performance in our case.

Reply:
Okay